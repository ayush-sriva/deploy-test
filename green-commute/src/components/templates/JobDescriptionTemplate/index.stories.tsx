import { ComponentMeta, ComponentStory } from '@storybook/react';
import React from 'react';
import JobDescriptionTemplate from '.';
import HP from '../../../../public/assets/logos/hp.png'


const description =
  'Open Text is seeking a talented, personable interaction designer who can assist the User Experience Design team by working with other designers and development teams on a variety of projects. The OpenText User Experience Design group is a distributed multi-disciplinary team of professionals that are responsible for enhancing the UX of the companyâ€™s collective product suites worldwide.';
const aboutCompany =
  'High level of proficiency with leading UX Design software packages, such as Axure, Sketch, InVision, or Experience Design including the core Adobe Creative Suite products.';
  const data={
    company: 'HP',
    title: 'User Experience Designer',
    location: 'Hyderabad, Telangana, India',
    companyLogo: HP,
    postedDate: '36min ago'
   }

export default {
  title: 'components/Templates/JobDescriptionTemplate',
  component: JobDescriptionTemplate,
} as ComponentMeta<typeof JobDescriptionTemplate>;

export const Template: ComponentStory<typeof JobDescriptionTemplate> = () => (
    <JobDescriptionTemplate description={description} aboutCompany={aboutCompany} currentLocation={'Hyderabad'} destination={'Mumbai'} jobDescData={data}/>
);
