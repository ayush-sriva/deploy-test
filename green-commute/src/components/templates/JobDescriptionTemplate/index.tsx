/**
 * @author Asish Kalintha <asish.sugun@zemosolabs.com>
 */

import { Grid, Paper, styled } from '@mui/material';
import responsiveTheme from '../../../theme';
import CommonRoutesCard from '../../organisms/CommonRoutes';
import CabCard from '../../organisms/CabCard';
import MetroCard from '../../organisms/MetroCard';
import Olaurl from '../../../../public/assets/logos/ola.png';
import Rapido from '../../../../public/assets/logos/rapido.png';
import Uber from '../../../../public/assets/logos/uber.png';
import JobDesCard from '../../organisms/JobDesCard';
import { useState } from 'react';
import JobDetailsCard from '../../organisms/JobDetailsCard';

interface JobData {
  company: string;
  title: string;
  location: string;
  companyLogo: string;
  postedDate: string;
}

interface JobDescriptionProps {
  description: string;
  aboutCompany: string;
  currentLocation: string;
  destination: string;
  jobDescData: JobData;
}
const MainContainer = styled(Paper)({
  maxWidth: '404px',
  paddingTop: '1.5rem',
  paddingLeft: '1.312rem',
  paddingRight: '1.312rem',
  height: '44.18rem',
  borderRadius: '0.75rem',
});

const Divider = styled('hr')({
  width: '24.8rem',
  border: `0.5px solid ${responsiveTheme.palette.greyScale.main}`,
  bordeRadius: '50%',
  marginTop: '1.5rem',
  marginBottom: '1.5rem',
});

const JobDescriptionTemplate: React.FC<JobDescriptionProps> = ({
  description,
  aboutCompany,
  currentLocation,
  destination,
  jobDescData,
}) => {
  const [jobDescription, setJobDescription] = useState(true);
  const [cabCard, setCabCard] = useState(false);

  const data = [
    {
      id: 'CabData1001',
      logoUrl: Olaurl,
      company: 'Ola',
      approx: 'Approximately  ',
      fare: '45',
    },
    {
      id: 'CabData1002',
      logoUrl: Rapido,
      company: 'Ola',
      approx: 'Approximately  ',
      fare: '45',
    },
    {
      id: 'CabData1003',
      logoUrl: Uber,
      company: 'Ola',
      approx: 'Approximately  ',
      fare: '45',
    },
  ];
  const toggleCommuteRoutes = () => {
    setJobDescription(!jobDescription);
  };
  const toggleRouteCards = () => {
    setCabCard(!cabCard);
    console.log('in toogleRouteCards');
  };

  const commonRoutesToggler = () => {
    if (cabCard) {
      return <CabCard data={data} />;
    } else {
      return <MetroCard />;
    }
  };
  const jobDescriptionToggler = () => {
    if (jobDescription) {
      return (
        <Grid item>
          <JobDetailsCard description={description} aboutCompany={aboutCompany} onClickHandler={toggleCommuteRoutes} />
        </Grid>
      );
    } else {
      return (
        <Grid item>
          <CommonRoutesCard
            currentLocation={currentLocation}
            destination={destination}
            onClickHandler={toggleCommuteRoutes}
            routeToggle={toggleRouteCards}
          />
          <Divider />

          <Grid item>{commonRoutesToggler()}</Grid>
        </Grid>
      );
    }
  };

  return (
    <MainContainer>
      <Grid container direction='column'>
        <Grid item>
          <JobDesCard
            companyLogo={jobDescData.companyLogo}
            company={jobDescData.company}
            title={jobDescData.title}
            location={jobDescData.location}
            postedDate={jobDescData.postedDate}
          />
        </Grid>
        <Divider />
        {jobDescriptionToggler()}
      </Grid>
    </MainContainer>
  );
};

export default JobDescriptionTemplate;
