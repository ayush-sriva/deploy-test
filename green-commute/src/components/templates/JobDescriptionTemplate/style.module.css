.cardSelected{
    border-color:#77EDDF;
    border-width: 2px;
    border-style: solid;
}
