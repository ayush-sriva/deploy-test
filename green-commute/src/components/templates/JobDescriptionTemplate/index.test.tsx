import { render, screen } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import JobDescriptionTemplate from '.';
import HP from '../../../../public/assets/logos/hp.png'


const description =
  'Open Text is seeking a talented, personable interaction designer who can assist the User Experience Design team by working with other designers and development teams on a variety of projects. The OpenText User Experience Design group is a distributed multi-disciplinary team of professionals that are responsible for enhancing the UX of the company’s collective product suites worldwide.';
const aboutCompany =
  'High level of proficiency with leading UX Design software packages, such as Axure, Sketch, InVision, or Experience Design including the core Adobe Creative Suite products.';
  const data={
    company: 'HP',
    title: 'User Experience Designer',
    location: 'Hyderabad, Telangana, India',
    companyLogo: HP,
    postedDate: '36min ago'
   }

function renderJobDescriptionTemplate() {
  render( <JobDescriptionTemplate description={description} aboutCompany={aboutCompany} currentLocation={'Hyderabad'} destination={'Mumbai'}  jobDescData={data}/>);
}

describe('JobDescriptionTemplate', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<JobDescriptionTemplate description={description} aboutCompany={aboutCompany} currentLocation={'Hyderabad'} destination={'Mumbai'}  jobDescData={data}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Text should render correctly', () => {
    renderJobDescriptionTemplate();
    const textElement = screen.getByText('Open Text is seeking a talented, personable interaction designer who can assist the User Experience Design team by working with other designers and development teams on a variety of projects. The OpenText User Experience Design group is a distributed multi-disciplinary team of professionals that are responsible for enhancing the UX of the company’s collective product suites worldwide.');
    expect(textElement).toBeInTheDocument();
  });
  test('Text should render correctly', () => {
    renderJobDescriptionTemplate();
    const textElement = screen.getByText('Hyderabad, Telangana, India');
    expect(textElement).toBeInTheDocument();
  });
  test('Number of Icons should be 2', () => {
    renderJobDescriptionTemplate();
    const textElement = screen.getAllByAltText('company_logo');
    console.log(textElement);
    expect(textElement.length).toBe(1);
  });
  test('Button element rendered correctly', () => {
    renderJobDescriptionTemplate();
    const ButtonElement = screen.getByTestId('Button1001');
    expect(ButtonElement).toBeInTheDocument();
  });
});
