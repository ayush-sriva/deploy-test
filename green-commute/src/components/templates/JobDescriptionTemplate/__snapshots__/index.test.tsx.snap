// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobDescriptionTemplate renders correctly 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-l8w7le-MuiPaper-root"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1vpwcmr-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiCard-root css-phpnom-MuiPaper-root-MuiCard-root"
      >
        <div
          className="MuiCardHeader-root css-so20wf-MuiCardHeader-root"
        >
          <div
            className="MuiCardHeader-avatar css-1ssile9-MuiCardHeader-avatar"
          >
            <img
              alt="company_logo"
              className="css-dejdd"
              src="hp.png"
            />
          </div>
          <div
            className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
          >
            <span
              className="MuiTypography-root MuiTypography-body2 MuiCardHeader-title css-et1ao3-MuiTypography-root"
            >
              <div
                className="MuiBox-root css-0"
              >
                <h6
                  className="MuiTypography-root MuiTypography-subtitle1 makeStyles-title-1 css-10wpov9-MuiTypography-root"
                >
                  User Experience Designer
                </h6>
                <span
                  className="MuiTypography-root MuiTypography-caption makeStyles-location-3 css-1sn4lm3-MuiTypography-root"
                >
                  HP
                </span>
              </div>
            </span>
            <span
              className="MuiTypography-root MuiTypography-body2 MuiCardHeader-subheader css-83ijpv-MuiTypography-root"
            >
              <div
                className="MuiBox-root css-0"
              >
                <span
                  className="MuiTypography-root MuiTypography-caption makeStyles-location-3 css-1sn4lm3-MuiTypography-root"
                >
                  Hyderabad, Telangana, India
                </span>
                <br />
                <span
                  className="MuiTypography-root MuiTypography-captionBold makeStyles-location-3 css-37jdci-MuiTypography-root"
                >
                  36min ago
                </span>
              </div>
            </span>
          </div>
          <div
            className="MuiCardHeader-action css-sgoict-MuiCardHeader-action"
          >
            <div
              className="css-u1w75p"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-yoei7b-MuiSvgIcon-root-MuiSvgIcon-root"
                data-testid="SVGIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="MuiCardActions-root MuiCardActions-spacing css-vb1tq8-MuiCardActions-root"
        >
          <button
            className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root  css-1k23hlb-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiTypography-root MuiTypography-captionBold css-37jdci-MuiTypography-root"
            >
              Save
            </span>
          </button>
          <button
            className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButtonBase-root  css-11qr2p8-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            <span
              className="MuiTypography-root MuiTypography-captionBold css-37jdci-MuiTypography-root"
            >
              Apply
            </span>
          </button>
        </div>
      </div>
    </div>
    <hr
      className="css-1jtvw9i"
    />
    <div
      className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-1320j2h-MuiPaper-root"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1vpwcmr-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item css-wtzyda-MuiGrid-root"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Description
            </p>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item css-63rm72-MuiGrid-root"
          >
            <p
              className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              Open Text is seeking a talented, personable interaction designer who can assist the User Experience Design team by working with other designers and development teams on a variety of projects. The OpenText User Experience Design group is a distributed multi-disciplinary team of professionals that are responsible for enhancing the UX of the companyâ€™s collective product suites worldwide.
            </p>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item css-wtzyda-MuiGrid-root"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              About Company
            </p>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item css-63rm72-MuiGrid-root"
          >
            <p
              className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              High level of proficiency with leading UX Design software packages, such as Axure, Sketch, InVision, or Experience Design including the core Adobe Creative Suite products.
            </p>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item css-63rm72-MuiGrid-root"
          >
            <p
              className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              Excellent written and oral communication and presentation skills
              <span>
                 
                <button
                  className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeLarge MuiButton-textSizeLarge MuiButtonBase-root  css-kpm0tx-MuiButtonBase-root-MuiButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="submit"
                >
                  <p
                    className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    SEE MORE
                  </p>
                </button>
                 
              </span>
            </p>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item css-vqb5yb-MuiGrid-root"
            data-testid="Button1001"
          >
            <button
              className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1hge7y5-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Green Commute Routes
              <span
                className="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  color="#30A193"
                  data-testid="ArrowForwardIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
